pandas/io/pytables.py

```diff
@@ -1656,7 +1657,8 @@ def convert(self, values, nan_rep, encoding, errors):
             kwargs['name'] = _ensure_decoded(self.index_name)
         try:
             self.values = Index(values, **kwargs)
-        except:
+        except (AttributeError, OverflowError, KeyError, IndexError,
```

Comment:
since this is so many things, ok here with just Exception (put a ``# noqa: E722`` to disable the warning)